### ISSUE TITLE
Live upload form with JS events inside LiveComponent targets parent LiveView

### ISSUE THREAD

**Issue by miguel-s** [phoenixframework/phoenix_live_view#1893]
### Environment

* Elixir version (elixir -v): Elixir 1.13.0 (compiled with Erlang/OTP 24)
* Phoenix version (mix deps): {:phoenix, "~> 1.6.6"}
* Phoenix LiveView version (mix deps): {:phoenix_live_view, "~> 0.17.5"}
* Operating system: macOS 12.2
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

Sample app: https://github.com/miguel-s/lv-upload-js-bug

Inside a LiveComponent, if instead of defining the form events in the standard way like so:
`<form id="test-upload-form" phx-change="validate-file" phx-submit="save-file" phx-target={@myself}>`

We try to use the new JS feature for pushing events, like so:
`<form id="test-upload-js-form" phx-change={JS.push("validate-file-js", target: @myself)} phx-submit={JS.push("save-file-js", target: @myself)}>`

The live upload will send the event to the parent LiveView instead of the LiveComponent.

Adding `phx-target={@myself}` fixes the issue, but seems redundant or confusing as we already specified the target inside the JS event:
`<form id="test-upload-js-form" phx-change={JS.push("validate-file-js", target: @myself)} phx-submit={JS.push("save-file-js", target: @myself)} phx-target={@myself}>`

#### Error

> [error] GenServer #PID<0.566.0> terminating
** (ArgumentError) no uploads have been allowed on LiveView named PlaygroundWeb.LiveUploadTest
    (phoenix_live_view 0.17.7) lib/phoenix_live_view/upload.ex:185: Phoenix.LiveView.Upload.get_upload_by_ref!/2
    (phoenix_live_view 0.17.7) lib/phoenix_live_view/upload.ex:120: Phoenix.LiveView.Upload.update_progress/4
    (phoenix_live_view 0.17.7) lib/phoenix_live_view/channel.ex:145: anonymous fn/4 in Phoenix.LiveView.Channel.handle_info/2
    (phoenix_live_view 0.17.7) lib/phoenix_live_view/channel.ex:1139: Phoenix.LiveView.Channel.write_socket/4
    (phoenix_live_view 0.17.7) lib/phoenix_live_view/channel.ex:143: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 3.16.1) gen_server.erl:695: :gen_server.try_dispatch/4
    (stdlib 3.16.1) gen_server.erl:771: :gen_server.handle_msg/6
    (stdlib 3.16.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3

### Expected behavior

The form should target the LiveComponent instead of the parent LiveView and allow the upload.

### Related issue
https://github.com/surface-ui/surface/issues/566

**Comment by janwillemvd** [comment]
Hi @miguel-s, thanks. We're having the same issue. We'll apply the fix like you suggested (adding `phx-target={@myself}`).

**Comment by dvic** [comment]
Just to add to this: I think this problem also occurs when you have a liveview inside a liveview.

**Comment by gushonorato** [comment]
I'm facing the same issue using uploads inside live component in old fashion, that means, without JS.push indirection. Adding `phx-target={@myself}` fixes it.

**Comment by chrismccord** [comment]
I cannot recreate this on LiveView master. Can you try latest and report back? Thanks!

**Comment by miguel-s** [comment]
Hi @chrismccord!

I can still reproduce this issue on LV master with the example app https://github.com/miguel-s/lv-upload-js-bug.

Elixir/OTP version:
> Erlang/OTP 24 [erts-12.1.5] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit] [dtrace]
> IEx 1.13.0 (compiled with Erlang/OTP 24)
> Interactive Elixir (1.13.0)

Mix deps:

```elixir
defp deps do
  [
    {:phoenix, "~> 1.6.6"},
    {:phoenix_html, "~> 3.0"},
    {:phoenix_live_reload, "~> 1.2", only: :dev},
    {:phoenix_live_view, github: "phoenixframework/phoenix_live_view", branch: "master", override: true},
    {:floki, ">= 0.30.0", only: :test},
    {:phoenix_live_dashboard, "~> 0.6"},
    {:esbuild, "~> 0.3", runtime: Mix.env() == :dev},
    {:swoosh, "~> 1.3"},
    {:telemetry_metrics, "~> 0.6"},
    {:telemetry_poller, "~> 1.0"},
    {:gettext, "~> 0.18"},
    {:jason, "~> 1.2"},
    {:plug_cowboy, "~> 2.5"}
  ]
end
```

Can I help with any additional info?

**Comment by joshchernoff** [comment]
I ran into a similar issue with this but because I was following the examples in the  cancel_upload/3 docs https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#cancel_upload/3 where I have my 
```
<button phx-click="cancel-upload" phx-value-ref={entry.ref} aria-label="cancel">&times;</button>
```
in a form who is rendered by a LiveComponent that was made by the default live generators. 

I was bewildered after I implemented 
```
  @impl Phoenix.LiveView
  def handle_event("cancel-upload", %{"ref" => ref, "value" => _}, socket) do
    {:noreply, cancel_upload(socket, :image, ref)}
  end
```
on my LiveComponent because it was failing.  I had not noticed till I read the logs better that it was failing to find the function on the parent liveview module not the live component that I had enabled uploads on.  So, it tried to set the phx-target to myself but that only resulted in firing off the form.
